_example/servers/sp-server/sp-server.go

```diff
@@ -112,18 +111,18 @@ func main() {
 		},
 	}
 
-	middleware := sp.NewMiddleware(&serviceProvider)
-
 	r := chi.NewRouter()
 	r.Use(logHandler)
 
-	r.Get(metadataPath, middleware.ServeMetadata)
-	r.Post(acsPath, middleware.ServeAcs(accessGranted))
+	r.Get(metadataPath, serviceProvider.MetadataHandler)
+
+	r.With(serviceProvider.AssertionConsumer).
```

Comment:
As suggested here: https://github.com/goware/saml/issues/2#issuecomment-337983678

Reply:
how about `r.With(sp.AssertionParser)` ?

Reply:
or `sp.IdPResponseParser`

Reply:
After reading https://github.com/goware/saml/pull/3#discussion_r147279930 I think we can call this either `ResponseHandler` or `AssertionHandler`, I prefer not using `Parser` here because I think we're not strictly parsing the message.

Reply:
Except that this is not a handler either, hah. It's a middleware, right? so `ResponseMiddleware` or `ProcessAssertion`, `AssertionProcessor`, `ResponseConsumer`? This part consumes a message and extracts some parts out of it.

Reply:
AssertionMiddleware is imho OK